### DISCUSSION TITLE
What is the difference between useFetch and useAsyncData??

### DISCUSSION THREAD

**Discussion by harblaith7** [nuxt/nuxt#18193] [Questions] [answered]
Based on the documentation, it says that:

**useAsyncData** can be used within your pages, components, and plugins to get access to data that resolves asynchronously

and 

**useFetch** can be used within your pages, components, and plugins to get universally fetch from any URL.

I am a little confused by this. So useFetch is not for data that resolvers asynchronously? useAsyncData doesn't fetch data from any url?

**Comment by wobsoriano** [comment, marked as answer]
`useAsyncData` accepts any function that returns a promise.

You might understand it better with an example.

Say you want to fetch todos using `useAsyncData`:

```html
<script setup lang="ts">
// todos here can be whatever. It's a unique key to ensure that data fetching can be properly de-duplicated across requests.
const { data } = useAsyncData('todos', () =>  $fetch('https://jsonplaceholder.typicode.com/todos'))
</script>
```

and using `useFetch`:

```html
<script setup lang="ts">
// a key is not required as it automatically generates a key based on url.
const { data } = useFetch('https://jsonplaceholder.typicode.com/todos')
</script>
```

**Reply by harblaith7** [reply]
Thanks, that is clearing it up. How would it work if you want to add additional logic in the function? Would it work like so?

```javascript
<script setup lang="ts">
const { data } = useAsyncData('todos', async () =>  {
      const theData = await $fetch('https://jsonplaceholder.typicode.com/todos');
      ... some logic ....
      return theData
})
</script>
```

**Reply by wobsoriano** [reply]
yeah that should work
